src/app/clusters/camera-av-settings-user-level-management-server/camera-av-settings-user-level-management-server.cpp

```diff
@@ -0,0 +1,1079 @@
+        {
+            handlerContext.mCommandHandler.AddStatus(handlerContext.mRequestPath, Status::UnsupportedCommand);
+        }
+        else
+        {
+            HandleCommand<Commands::MPTZSavePreset::DecodableType>(
+                handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandleMPTZSavePreset(ctx, commandData); });
+        }
+        return;
+
+    case Commands::MPTZRemovePreset::Id:
+        ChipLogDetail(Zcl, "CameraAVSettingsUserLevelMgmt: Remove indicated MPTZ Preset");
+
+        if (!HasFeature(Feature::kMechanicalPresets))
+        {
+            handlerContext.mCommandHandler.AddStatus(handlerContext.mRequestPath, Status::UnsupportedCommand);
+        }
+        else
+        {
+            HandleCommand<Commands::MPTZRemovePreset::DecodableType>(
+                handlerContext,
+                [this](HandlerContext & ctx, const auto & commandData) { HandleMPTZRemovePreset(ctx, commandData); });
+        }
+        return;
+
+    case Commands::DPTZSetViewport::Id:
+        ChipLogDetail(Zcl, "CameraAVSettingsUserLevelMgmt: Setting DPTZ Viewport");
+
+        if (!HasFeature(Feature::kDigitalPTZ))
+        {
+            handlerContext.mCommandHandler.AddStatus(handlerContext.mRequestPath, Status::UnsupportedCommand);
+        }
+        else
+        {
+            HandleCommand<Commands::DPTZSetViewport::DecodableType>(
+                handlerContext,
+                [this](HandlerContext & ctx, const auto & commandData) { HandleDPTZSetViewport(ctx, commandData); });
+        }
+        return;
+
+    case Commands::DPTZRelativeMove::Id:
+        ChipLogDetail(Zcl, "CameraAVSettingsUserLevelMgmt: Relative move within a defined DPTZ Viewport");
+
+        if (!HasFeature(Feature::kDigitalPTZ))
+        {
+            handlerContext.mCommandHandler.AddStatus(handlerContext.mRequestPath, Status::UnsupportedCommand);
+        }
+        else
+        {
+            HandleCommand<Commands::DPTZRelativeMove::DecodableType>(
+                handlerContext,
+                [this](HandlerContext & ctx, const auto & commandData) { HandleDPTZRelativeMove(ctx, commandData); });
+        }
+        return;
+    }
+}
+
+void CameraAvSettingsUserLevelMgmtServer::HandleMPTZSetPosition(HandlerContext & ctx,
+                                                                const Commands::MPTZSetPosition::DecodableType & commandData)
+{
+    Status status           = Status::Success;
+    bool hasAtLeastOneValue = false;
+
+    Optional<int16_t> pan  = commandData.pan;
+    Optional<int16_t> tilt = commandData.tilt;
+    Optional<uint8_t> zoom = commandData.zoom;
+
+    // Validate the received command fields
+    //
+    if (pan.HasValue())
+    {
+        if (!HasFeature(Feature::kMechanicalPan))
+        {
+            ChipLogError(Zcl, "Mechanical Pan not supported although pan value provided");
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure);
```

Comment:
```suggestion
            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::InvalidCommand);
```
is what the spec says, no?